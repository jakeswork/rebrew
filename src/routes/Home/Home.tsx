import React, { Fragment } from "react";

import { Beer } from "../../types";

interface HomeProps {
  beers: Beer[];
}

const Home: React.FC<HomeProps> = ({ beers }) => (
  <Fragment>
    <h1>We've got some beers!</h1>
    {beers.map(beer => (
      <div key={beer.id}>
        <h4>{beer.name}</h4>
        <h5>{beer.description}</h5>
      </div>
    ))}
  </Fragment>
);

export default Home;
