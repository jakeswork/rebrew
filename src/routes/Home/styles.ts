import backgroundImg from "../../images/rebrew-bg-2.jpg";

export default {
  root: {
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    backgroundImage: `url(${backgroundImg})`,
    height: "100vh",
    display: "flex",
    alignItems: "center"
  },
  heading: {
    width: 440,
    marginLeft: 64
  },
  searchWrapper: {
    display: "flex",
    justifyContent: "space-between"
  }
};
