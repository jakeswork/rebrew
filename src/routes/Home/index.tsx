import React, { ReactElement } from "react";
import withStyles from "react-jss";
import { useQuery } from "@apollo/react-hooks";
import { gql } from "apollo-boost";

import Home from "./Home";
import styles from "./styles";

const gqlQuery: string = gql`
  {
    beers {
      id
      name
      description
    }
  }
`;
const Composed: React.FC = (): ReactElement => {
  const { loading, error, data } = useQuery(gqlQuery);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  return <Home {...data} />;
};

export default withStyles(styles)(Composed);
