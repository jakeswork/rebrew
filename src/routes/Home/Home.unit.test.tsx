import React from "react";
import { shallow } from "enzyme";

import Home from "./Home";

const defaultProps = {
  classes: {}
};

describe("The Home instance", () => {
  beforeEach(() => jest.clearAllMocks());

  describe("When rendered with default props", () => {
    it("Should render without throwing any errors", () => {
      const component = shallow(<Home {...defaultProps} />);

      console.error = jest.fn();

      expect(component).toMatchSnapshot();
      expect(console.error).not.toHaveBeenCalled();
    });
  });
});
