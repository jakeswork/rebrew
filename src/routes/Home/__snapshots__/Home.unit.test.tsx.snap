// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Home instance When rendered with default props Should render without throwing any errors 1`] = `ShallowWrapper {}`;
